app/submitter/converter_v2.py

```diff
@@ -123,15 +123,39 @@ def get_payload_data(
             schema=schema,
             full_routing_path=routing_path,
         )
+
     if schema.json["data_version"] == "0.0.3":
-        return {
-            "answers": convert_answers_to_payload_0_0_3(
-                answer_store=answer_store,
-                list_store=list_store,
-                schema=schema,
-                full_routing_path=routing_path,
-            ),
+        answers = convert_answers_to_payload_0_0_3(
+            answer_store=answer_store,
+            list_store=list_store,
+            schema=schema,
+            full_routing_path=routing_path,
+        )
+
+        data: dict[str, Union[list[Any]]] = {
+            "answers": answers,
             "lists": list_store.serialize(),
         }
 
+        if answer_codes := schema.json.get("answer_codes"):
+            answer_ids_to_filter = {answer.answer_id for answer in answers}
+            if filtered_answer_codes := get_filtered_answer_codes(
+                answer_codes, answer_ids_to_filter
+            ):
+                data["answer_codes"] = filtered_answer_codes
+
+        return data
+
     raise DataVersionError(schema.json["data_version"])
+
+
+def get_filtered_answer_codes(
+    answer_codes: Iterable[dict], answer_ids_to_filter: set[str]
```

Comment:
I suggest enforcing keyword args so these can't be passed wrong way around.

Reply:
Enforcing now 👍